File: backend/NewLevel/Entities/Event.cs
using NewLevel.Enums;
using NewLevel.Enums.Event;

namespace NewLevel.Entities
{
    public class Event
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string? Description { get; set; }
        public DateTime DateStart { get; set; }
        public DateTime? DateEnd { get; set; }
        public string Location { get; set; }
        public string? bannerUrl { get; set; }
        public int OrganizerId { get; set; }
        public List<EMusicGenres> Genre { get; set; }
        public string? TicketsLink { get; set; }
        public decimal? Price { get; set; }
        public int? Capacity { get; set; }
        public EEventStatus EventStatus { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime? UpdatedAt { get; set; }
    }
}

File: frontend/src/types/Event.ts
import { EMusicGenres } from '../gen/api/src/models/EMusicGenres';
import { PhotoResponseDto } from '../gen/api/src/models/PhotoResponseDto';
import { BandDto } from '../gen/api/src/models/BandDto';
import { EventResponseDto } from '../gen/api/src';

export enum EEventStatus {
  Draft = 0,
  Published = 1,
  InProgress = 2,
  Completed = 3,
  Cancelled = 4
}

export interface EventDto {
  id?: number;
  title: string;
  description?: string;
  dateStart: string;
  dateEnd?: string;
  location: string;
  bannerUrl?: string;
  organizerId: number;
  genre: EMusicGenres[];
  ticketsLink?: string;
  price?: number;
  capacity?: number;
  eventStatus: EEventStatus;
  bands?: BandDto[];
  photos?: PhotoResponseDto[];
  commentsCount?: number;
  createdAt?: string;
  updatedAt?: string;
}

export interface EventResponseDtoGenericList {
  items?: EventResponseDto[];
  totalCount?: number;
}

export interface EventResponseDtoGenericListNewLevelResponse {
  data?: EventResponseDtoGenericList;
  isSuccess?: boolean;
  message?: string;
}
